### PR TITLE
refactor: make shell.ShowItemInFolder asynchronous

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

BREAKING CHANGE: change return type of `ShowItemInFolder` to `void` from `bool` and offload task to new thread.

cc @nornagon @deepak1556

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Made `ShowItemInFolder` asynchronous with no return value.

### CODE REVIEW

**Comment by deepak1556** [comment]
Oops missed that, yeah the main thread disallows blocking calls. We can't add [base::ScopedAllowBlockng](https://cs.chromium.org/chromium/src/base/threading/thread_restrictions.h?l=311) without patch to upstream. Best way forward is to move these tasks to a different thread, then again its not possible with synchronous api versions.

**Comment by deepak1556** [comment]
> also whats the benefit of these two apis returning synchronous boolean results ? If it can be deprecated for async version, we could offload the task to a different thread.

@zcbenz what are your thoughts on this ?

**Comment by zcbenz** [comment]
Deprecating sync versions looks good to me.

**Comment by codebytere** [comment]
@zcbenz @deepak1556 sounds good to me! I'll start working on that :)

File: atom/common/platform_util_win.cc [hunk 1]
```diff
@@ -231,63 +232,65 @@ HRESULT DeleteFileProgressSink::ResumeTimer() {
 
 namespace platform_util {
 
-bool ShowItemInFolder(const base::FilePath& full_path) {
+void ShowItemInFolderOnWorkerThread(const base::FilePath& full_path) {
```
Comment (deepak1556) [inline atom/common/platform_util_win.cc]: Move this helper function to an anonymous namespace

**Comment by nornagon** [comment]
NB. you can also write the line "BREAKING CHANGE" in the PR description to have it listed in that section of the release notes.

File: atom/common/platform_util_win.cc (hunk 1 above)
Comment (Gam3356) [reply atom/common/platform_util_win.cc:233]: Thank you guys! Also want to thank God for this financial blessing. Willie Parish

**Pull request merged**